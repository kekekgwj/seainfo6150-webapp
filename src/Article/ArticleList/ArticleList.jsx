import React from "react";
import ArticleListItem from "../../ArticleListItem/ArticleListItem.jsx";
import style from "./ArticleList.module.css";
const ArticleList = (props) => {
    const { articles } = props;

    return !articles ? ( <p>You have no data!</p> ) : 
    (<div className={style.articleListContainer}>
        <ul className= {style.articleList}>
        {Object.values(articles).map((article, index) => {
            return  (
                <li key={ index }>
                    <ArticleListItem article={ article }></ArticleListItem>
                </li>
            )
        })}
        </ul>
    </div>);
}

export default ArticleList;