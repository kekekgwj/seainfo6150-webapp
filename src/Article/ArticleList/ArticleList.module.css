.articleListContainer {
    background-color: grey;
    padding: 10px 0 30px 10px;

}
.articleList {
    list-style: none;
    display: grid;
    grid-template-columns: repeat(3, 33.33%);
    grid-row-gap: 50px;
    grid-column-gap: 20px;
}
ul {
   padding-right: 80px;
}
li {
    min-width: 150px;
    min-height: 160px;
}
@media (min-width: 1600px) {
    .articleList {
        display: flex;
        flex-wrap: wrap;
    }
}