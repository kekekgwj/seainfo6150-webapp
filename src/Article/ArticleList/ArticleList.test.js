import React from "react";
import ArticleList from "./ArticleList";

describe("ArticleList tests", () => {
    it("renders correctly", () => {
      const { container } = render(<ArticleList />);
      expect(container).toMatchSnapshot();
    });
})
