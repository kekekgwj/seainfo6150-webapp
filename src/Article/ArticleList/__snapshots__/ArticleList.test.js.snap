// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList tests renders correctly 1`] = `
<div>
  <p>
    You have no data!
  </p>
</div>
`;
